### ISSUE TITLE
`Debug` not implemented for `HkdfExtract<Sha256>` or `Hkdf<Sha256>`

### ISSUE THREAD

**Issue by codahale** [RustCrypto/KDFs#89]
Here’s a minimal test case:

```rust
#[test]
fn test_debug() {
    fn check_debug<T: core::fmt::Debug>() {}

    check_debug::<HkdfExtract<Sha256>>();
    check_debug::<Hkdf<Sha256>>();
}
```

It fails with this error:

```text
error[E0277]: the trait bound `CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>: AlgorithmName` is not satisfied
   --> hkdf/tests/tests.rs:388:19
    |
388 |     check_debug::<HkdfExtract<Sha256>>();
    |                   ^^^^^^^^^^^^^^^^^^^ the trait `AlgorithmName` is not implemented for `CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>`
    |
    = help: the following other types implement trait `AlgorithmName`:
              HmacCore<D>
              Sha1Core
              Sha256VarCore
              Sha512VarCore
              CtVariableCoreWrapper<T, OutSize, O>
    = note: required for `HkdfExtract<CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>>` to implement `Debug`
note: required by a bound in `check_debug`
   --> hkdf/tests/tests.rs:386:23
    |
386 |     fn check_debug<T: core::fmt::Debug>() {}
    |                       ^^^^^^^^^^^^^^^^ required by this bound in `check_debug`

error[E0277]: the trait bound `CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>: AlgorithmName` is not satisfied
   --> hkdf/tests/tests.rs:389:19
    |
389 |     check_debug::<Hkdf<Sha256>>();
    |                   ^^^^^^^^^^^^ the trait `AlgorithmName` is not implemented for `CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>`
    |
    = help: the following other types implement trait `AlgorithmName`:
              HmacCore<D>
              Sha1Core
              Sha256VarCore
              Sha512VarCore
              CtVariableCoreWrapper<T, OutSize, O>
    = note: required for `Hkdf<CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>>` to implement `Debug`
note: required by a bound in `check_debug`
   --> hkdf/tests/tests.rs:386:23
    |
386 |     fn check_debug<T: core::fmt::Debug>() {}
    |                       ^^^^^^^^^^^^^^^^ required by this bound in `check_debug`

For more information about this error, try `rustc --explain E0277`.
```

Any ideas?

(Edited with a more concise test case and error message.)

**Comment by newpavlov** [comment]
The generic `Debug` impls for `hkdf` types are written not quite correctly. I forgot that we do not implement the `AlgorithmName` trait for "wrapped" types (e.g. `Sha256`) only for "core" types. We probably should fix it in the upstream crates, but for now I will push a quick fix.

**Comment by newpavlov** [comment]
Fix released in `hmac` v0.12.4.

**Comment by codahale** [comment]
Awesome, thank you!
